cpp/oneapi/dal/algo/triangle_counting/test/triangle_counting_test.cpp

```diff
@@ -0,0 +1,394 @@
+    std::array<std::int32_t, 6> cols = { 1, 2, 0, 2, 0, 1 };
+    std::array<std::int64_t, 4> rows = { 0, 2, 4, 6 };
+    std::array<std::int64_t, 3> local_triangles = { 1, 1, 1 };
+};
+
+class wheel_graph_6_type : public graph_base_data {
+public:
+    wheel_graph_6_type() {
+        vertex_count = 6;
+        edge_count = 10;
+        global_triangle_count = 5;
+    }
+
+    std::array<std::int32_t, 6> degrees = { 5, 3, 3, 3, 3, 3 };
+    std::array<std::int32_t, 20> cols = {
+        1, 2, 3, 4, 5, 0, 2, 5, 0, 1, 3, 0, 2, 4, 0, 3, 5, 0, 1, 4
+    };
+    std::array<std::int64_t, 7> rows = { 0, 5, 8, 11, 14, 17, 20 };
+    std::array<std::int64_t, 6> local_triangles = { 5, 2, 2, 2, 2, 2 };
+};
+
+class isolated_vertices_graph_10_type : public graph_base_data {
+public:
+    isolated_vertices_graph_10_type() {
+        vertex_count = 10;
+        edge_count = 11;
+        global_triangle_count = 5;
+    }
+
+    std::array<std::int32_t, 10> degrees = { 5, 3, 2, 0, 3, 4, 0, 2, 0, 3 };
+    std::array<std::int32_t, 22> cols = { 1, 2, 4, 5, 7, 0, 5, 9, 0, 7, 0,
+                                          5, 9, 0, 1, 4, 9, 0, 2, 1, 4, 5 };
+    std::array<std::int64_t, 11> rows = { 0, 5, 8, 10, 10, 13, 17, 17, 19, 19, 22 };
+    std::array<std::int64_t, 10> local_triangles = { 3, 2, 1, 0, 2, 4, 0, 1, 0, 2 };
+};
+
+class isolated_vertex_graph_11_type : public graph_base_data {
+public:
+    isolated_vertex_graph_11_type() {
+        vertex_count = 11;
+        edge_count = 45;
+        global_triangle_count = 120;
+    }
+
+    std::array<std::int32_t, 11> degrees = { 10, 10, 10, 10, 10, 0, 10, 10, 10, 10, 10 };
+    std::array<std::int32_t, 90> cols = { 1, 2, 3, 4, 6, 7, 8, 9, 10, 0, 2, 3, 4, 6, 7, 8, 9, 10,
+                                          0, 1, 3, 4, 6, 7, 8, 9, 10, 0, 1, 2, 4, 6, 7, 8, 9, 10,
+                                          0, 1, 2, 3, 6, 7, 8, 9, 10, 0, 1, 2, 3, 4, 7, 8, 9, 10,
+                                          0, 1, 2, 3, 4, 6, 8, 9, 10, 0, 1, 2, 3, 4, 6, 7, 9, 10,
+                                          0, 1, 2, 3, 4, 6, 7, 8, 10, 0, 1, 2, 3, 4, 6, 7, 8, 9 };
+
+    std::array<std::int64_t, 12> rows = { 0, 9, 18, 27, 36, 45, 45, 54, 63, 72, 81, 90 };
+    std::array<std::int64_t, 11> local_triangles = { 36, 36, 36, 36, 36, 0, 36, 36, 36, 36, 36 };
+};
+
+class triangle_counting_test {
+public:
+    using my_graph_type = dal::preview::undirected_adjacency_vector_graph<>;
+
+    template <typename GraphType>
+    auto create_graph() {
+        GraphType graph_data;
+        my_graph_type my_graph;
+
+        auto &graph_impl = oneapi::dal::detail::get_impl(my_graph);
+        auto &vertex_allocator = graph_impl._vertex_allocator;
+        auto &edge_allocator = graph_impl._edge_allocator;
+
+        const std::int64_t vertex_count = graph_data.get_correct_vertex_count();
+        const std::int64_t edge_count = graph_data.get_correct_edge_count();
+        const std::int64_t cols_count = edge_count * 2;
```

Comment:
Can you store it in the graph class? Without magic x2 which is related to the concrete representation of an undirected graph.
That is cols_count = graph_data.get_cols_count()